include/ocpp/common/evse_security.hpp

```diff
@@ -87,18 +87,19 @@ class EvseSecurity {
     /// \param organization
     /// \param common
     /// \return the PEM formatted certificate signing request
-    virtual std::string generate_certificate_signing_request(const CertificateSigningUseEnum& certificate_type,
-                                                             const std::string& country,
-                                                             const std::string& organization, const std::string& common,
-                                                             bool use_tpm) = 0;
+    virtual std::optional<std::string>
```

Comment:
Rather than using std::optional I would like to see the `GetCertificateSignRequestResult` type here as well. This transmits more information about a possible failure to create the CSR. This information can be used to send a SecurityEventNotification to the CSMS.

Reply:
Understood, will be updated.